galley/pkg/mcp/server/server.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+	"sync/atomic"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
```

Comment:
We should not introduce dependency on go-control-plane.

The ads.proto is here:
https://github.com/envoyproxy/data-plane-api/blob/master/envoy/service/discovery/v2/ads.proto

Reply:
I see in fact it is already used in pilot, I am a bit confused.

Reply:
I used xDS for initial development. See https://github.com/istio/api/pull/552 for the derived MCP service and message definitions.